bazel/example/main.cc

```diff
@@ -9,7 +9,8 @@
 using ::google::protobuf::TextFormat;
 using ::p4::config::v1::P4Info;
 
-using namespace P4::literals;
+using namespace ::P4;
+using namespace ::P4::literals;
```

Comment:
Why is the global prefix (`::`) needed?

Reply:
I was mostly following the reasoning in https://abseil.io/tips/119 to use fully qualified names. This doc discusses using-declaration, not `using namespace` as it is discouraged to be used at all in the [Google C++ Style Guide](https://google.github.io/styleguide/cppguide.html#Namespaces), but I feel the reasoning still applies.

However, as it's very unlikely that we will ever introduce any nested inner `P4` namespace, I don't see dropping the `::` prefix would cause any issue. So if that's the preferred coding style, I can remove the `::` prefixes.

Reply:
Converted back to `using namespace P4` to be more consistent with existing code.